jax/_src/config.py

```diff
@@ -546,7 +546,7 @@ def update_thread_local_jit_state(**kw):
 config.define_enum_state(
     name='jax_default_dtype_bits',
     enum_values=['32', '64'],
-    default='64',
+    default='32',
```

Comment:
What is the impact of this change?

Reply:
It makes the default dtype 32-bit. So, for example, with `jax_enavle_x64=True`, `jnp.arange(10)` would return `int32` instead of `int64`.

Reply:
I'm preparing a JEP/design doc that will discuss all this; hopefully will open an issue soon. But this is part of a change we've been discussing and working toward since mid-2020.

Reply:
Will you wait for this design doc to be public before merging this change?

Reply:
Yeah, this PR mainly exists for testing purposes. I doubt it will be merged.